src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -498,7 +498,9 @@ public function dumpFile($filename, $content, $mode = 0666)
 
         if (!is_dir($dir)) {
             $this->mkdir($dir);
-        } elseif (!is_writable($dir)) {
+        }
+
+        if (!is_writable($dir)) {
```

Comment:
this check is already bundled into file_put_contents' failure handling. You're just making this case more explicit, but file_put_contents can fail for more reasons.
So, should it be considered a new feature? Should we use error_get_last to generate a more useful error message? should this go on master?